src/librustc_mir/tcx/expr.rs

```diff
@@ -271,44 +287,47 @@ impl<'tcx> Mirror<'tcx> for &'tcx hir::Expr {
             }
 
             // Now comes the rote stuff:
-
-            hir::ExprRepeat(ref v, ref c) =>
-                ExprKind::Repeat { value: v.to_ref(), count: c.to_ref() },
-            hir::ExprRet(ref v) =>
-                ExprKind::Return { value: v.to_ref() },
-            hir::ExprBreak(label) =>
-                ExprKind::Break { label: label.map(|_| loop_label(cx, self)) },
+            hir::ExprRepeat(ref v, ref c) => ExprKind::Repeat {
+                value: v.to_ref(),
+                count: c.to_ref(),
+            },
+            hir::ExprRet(ref v) => ExprKind::Return { value: v.to_ref() },
+            hir::ExprBreak(label) => ExprKind::Break { label: label.map(|_| loop_label(cx, self)) },
             hir::ExprAgain(label) =>
                 ExprKind::Continue { label: label.map(|_| loop_label(cx, self)) },
```

Comment:
Hmm, not clear to me either. We have a heuristic which pushes struct lits to multi-line mode after a fairly short inner width is reached, but in this case the preserved struct lit looks longer than the broken one. I think we might not break if there is only one field. We could do better with some non-local reasoning here.